strongbox-web-core/src/main/java/org/carlspring/strongbox/config/SecurityConfig.java

```diff
@@ -53,6 +55,8 @@ protected void configure(HttpSecurity http)
         http.sessionManagement()
             .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
             .and()
+            .exceptionHandling().authenticationEntryPoint(basicAuthenticationEntryPoint())
```

Comment:
Right now it fits your requirement to ask user for username/password as HttpBasic states. I think we can handle your improvements in the following tasks. 

@carlspring, what do you think ?